docs/machine.md

```diff
@@ -0,0 +1,30 @@
+# Using with Docker Toolbox and Machine
+
+In development, we recommend using [Docker Toolbox](https://www.docker.com/products/docker-toolbox) to set up Docker. It includes a tool called Machine which will create a VM running Docker Engine and point your shell at it using environment variables.
```

Comment:
To be clear: the vast majority of **first-time users in development** are going to have the best experience if we recommend using Toolbox, so we put that front and centre. For users doing something more complex, they are probably knowledgeable enough that they don't need this handholding.